### PR TITLE
Fix and update ICU calculation and methodology text.

### PR DESCRIPTION
**Pull request by artdgn**
Following @HenkPoley's [comment](https://github.com/github/covid19-dashboard/commit/0ac1262c8f81b9696556fb3b21d71ba533f198bb#commitcomment-38361446) I've fixed the logical mistake with the ICU calculation (fractions are hard), updated the constant used, and also updated the method of calculation (to be a ratio of the reported active cases and not a ratio of the estimated cases).
The table looks considerably redder and worrisome now :(

### CODE REVIEW

**Comment by HenkPoley** [comment]
I notice the model is still seems fairly far off for anything about The Netherlands.

We discover about a 1000 cases per day. Over 5 days that would be 5000. This model expects 58546. ~11x more.

From China's 45K cases report 13.8% severe + 4.7% critical = 18.5%. Let's assume we only find those in The Netherlands, everybody straight to hospital. That means there were 5k/0.185 ~= 27k total infected ~10 days ago. 

So even with a strong under counting assumption, the model here predicts ~2x more.

We currently have ~150 deaths a day. 5 days * 150 people / 0.045 ~= 17k total infected ?? days ago. 

So with that assumption, the model here predicts ~3.5x more _active_ cases.

Currently the counts in The Netherlands seem to be plateauing / in a steady state. Maybe the model is forced to assume a stronger growth?

btw, potential ICU capacity has been scaled up to 2400 beds. Since we went past the 1150 beds on the 31st of March. And if you keep in mind the 575 reserved for 'normal' patients, we went over the remaining beds on the 23rd, and the COVID-19 patients section has a capacity of 2400-575 ~= 1800-1900. 

E.g. ~10.9 per 100k population.

**Comment by HenkPoley** [comment]
Also: If actually 1.5% of the country was infected we would have in the order of 800k deaths by now. We would know.

**Comment by hamelsmu** [comment]
FYI on a related note, I updated all dashboards that contain a predictive model with a more stern warning

image

**Comment by HenkPoley** [comment]
😅

**Comment by artdgn** [comment]
@HenkPoley thanks for that analysis. 

> We discover about a 1000 cases per day. Over 5 days that would be 5000. This model expects 58546. ~11x more.
So even with a strong under counting assumption, the model here predicts ~2x more.
We currently have ~150 deaths a day. 5 days * 150 people / 0.045 ~= 17k total infected ?? days ago.
So with that assumption, the model here predicts ~3.5x more active cases.

This is due to estimating the "testing bias" from deaths. The reported fatality rate is very high (unfortunately) at 10.7%, and the estimation of actual cases is trying to calculate the testing bias from deaths relative to lagged cases. In this case the fatality rate for current deaths over lagged cases (8 days ago): is 2256 / 12668 = 11.8%. It is than compared to the fatality rate that was found in countries where testing was comprehensive - 1.5%. And this ratio is 11.8 / 1.5 gives an estimate of testing ratio of 7.8. So this number is used to adjust the reported case numbers. While the exact numbers in this estimation might not be totallly correct, it seems reasonable that there is significant under counting if the fatality rate (relative to reported cases) is so high. 
These "total cases" are including the asymptomatic cases as well.

> Currently the counts in The Netherlands seem to be plateauing / in a steady state. R0 is estimated to wobble near 1 now (probably below). Maybe the model is forced to assume a stronger growth?

The growth rate is calculated from the last 5 days data, so if that data is reliable and consistent, this should reflect in the calculation. The growth rate I see in the table is 5.9%, which is consistent with nearly a plateau (since 5% is about the size of just the recovery rate), so active cases should be rising very slowly at this point (6% added daily, and 5% removed daily). Perhaps in a few days the projections will level out completely, when growth rate estimation catches up.

> epiforecasts 90% confidence max. is about 8300 new cases in 5 days: https://epiforecasts.io/covid/posts/national/netherlands/

The link was 404, but this https://epiforecasts.io/covid/ worked so I'll definitely have a look there.

> btw, potential ICU capacity has been scaled up to 2400 beds..

That's great. I imagine most countries' capacities are increasing. Not sure how to factor that into the capacity estimations.. and it's probably going to be more important as time progresses. Any ideas for data sources? Maybe just multiply by some constant factor, any ideas for data to estimate that "scaling" factor from?

> Also: If actually 1.5% of the country was infected we would have in the order of 12k deaths by now. We would know.

With the fatality rate assumed in this model (for asymptomatic cases included in totals) I get 17000000 x 0.015 (total assumed infected) x 0.015 (fatality rate) = 3825, which is not the current number of expected deaths, but the number expected in 8 days. With the current number of deaths being 2256, and the new deaths during last 5 days being 765, this might not be too far off unfortunately. Does this make sense? Or am I missing something about your calculations?

Thanks again for fact checking this vs reality!

**Comment by HenkPoley** [comment]
> 0.015 (fatality rate) 

Ah, your fatality rate estimate is rather low. I'm wonder though why/how would be missing about 11 out of 12 (91.3%) infected cases.

Maybe you are not taking into account that The Netherlands is on average older than China? I believe the scaling factor for the CFR for Greece could even be ~4x, just looking at the population pyramid.

>> btw, potential ICU capacity has been scaled up to 2400 beds..
>
> Maybe just multiply by some constant factor, any ideas for data to estimate that "scaling" factor from?

There will be no further scaling of ICU, it has already gone from 2 beds per medic to 4 beds per 1 medic. Unless of course over many many quarters (e.g. full but shortened education cycle for new intensivists).

>> epiforecasts 90% confidence max. is about 8300 new cases in 5 days: https://epiforecasts.io/covid/posts/national/netherlands/
> 
> The link was 404, but this https://epiforecasts.io/covid/ worked so I'll definitely have a look there.

Epiforecasts [somehow dropped](https://github.com/epiforecasts/covid/issues/31) The Netherlands overnight 🤷‍♂️.

**Comment by HenkPoley** [comment]
Re: scaling of CFR with population pyramid: https://twitter.com/CJEMetcalf/status/1229792572513669121

**Comment by artdgn** [comment]
@HenkPoley

> Ah, your fatality rate estimate is rather low. I'm wonder though why/how would be missing about 11 out of 12 (91.3%) infected cases. 

> Re: scaling of CFR with population pyramid: https://twitter.com/CJEMetcalf/status/1229792572513669121

Adjusting the expected CFR by country would be a great improvement, I agree. 
One point about the 1.5% CFR that I've used, is that it's not based on China's CFR, but instead on Cruise Ship, South Korea and Singapore at the time. Which are not that young, so I don't think it would be a factor of 3x / 4x but hopefully more around 2x (from the population-pyramid data). I think the main difference for these countries (bringing the CFR so low) was the fact that asymptomatic cases are accounted in it as well (more so than in countries that are testing only the suspected cases). To me, this is a more important measurement since for estimating future growth asymptomatic cases are very important.

> Just oggling CFR's here it seems to corroborate that: https://avatorl.org/covid-19/?page=CaseFatalityRate

That's a great resource, thanks! How do you find these?

**Comment by artdgn** [comment]
@HenkPoley 
Another point about CFR and testing bias relationship to age is perhaps to consider Japan and Germany, which have old populations, but in which the CFR is around 2%.

**Comment by HenkPoley** [comment]
> it's not based on China's CFR, but instead on Cruise Ship, South Korea and Singapore at the time. 

Do you have a reason to believe these people were all tracked properly, and not transferred to their respective home countries and thus 'lost' in the statistics ?

South Korea's naive CFR is not 1.5% (nor [Resolved CFR](http://blog.zorinaq.com/case-fatality-ratio-ncov/)). And it is still rising.

Singapore is still low, yes. But don't they have the majority of infections under travellers and foreign workers (both a healthy group) ?

> That's a great resource, thanks! How do you find these?

Loosely based on how Christopher Hitchens was reading newspapers. He would grab a paper and skip reading everything but the name of the reporters, and if he knew it was a good reporter he would read the article. With social media you don't need to do the first step, you just only/mostly follow the people who seem to know things, and seem to be able to filter out nonsense.

Maybe go down my most recent fresh followers 🤷‍♂️: https://twitter.com/henkpoley/following

I've studied Bio-informatics years ago, so I have some feeling with the field.

A resource that was good very early on, but not specifically for your statistics: http://virological.org/c/novel-2019-coronavirus/ncov-2019-genomic-epidemiology/36

People who gather (and link to) there are people who seem to know things.

At the PowerBI site (avatorl), the linked resource BNOnews is also a person who seems to know things.

There are also people who only _seem_ to know things, but who do not have proper filters. For COVID-19 specifically, Eric Feigl-Ding comes to mind. But people who cluster around Nassim Nicholas Taleb are also strong on this factor. Very right- and very left-wing politics is also smell for this.

Basically: I spend too much on Twitter 😋

**Comment by artdgn** [comment]
> Basically: I spend too much on Twitter yum

@HenkPoley Thanks for those resources, I definitely do not spend enough time on twitter, looks like a great source of info.. I did however find this [conventionally, although problematically, named list](https://github.com/soroushchehresa/awesome-coronavirus). 

About the true CFR. I find the age theory compelling, but difficult to reconcile with some counter examples:
1. The cruise ship population was fairly old, and [here they estimate](https://cmmid.github.io/topics/covid19/severity/diamond_cruise_cfr_estimates.html) the CFR (accounting for the lag) to be 2.3% (which is higher than what I'm using, and it might be worth adjusting it already to that figure).
2. In addition to the cruise ship data, it is hard to reconcile some additional counter examples of how the CFR can be so low in some countries. I'm calculating the lagged fatality rate with a lag of 8 days (that is also likely on top of the implicit testing lag of around 5 days and also the implicit likely lag of symptoms onset of another several days - so should be of 2-3 weeks). Indeed for South Korea it is 2.1%, Germany is 3.2% and Japan is at 4% but some countries are below 1%, and there are even more below 2%, and Australia, Israel, New Zealand are not particularly young countries, neither are they likely to misreport deaths:
```
                      Cases.total  Cases.new  Deaths.total  lagged_fatality_rate
Country/Region                                                                  
New Zealand                  1283        244             2              0.003759
Qatar                        2512        908             6              0.007368
Singapore                    2108        799             7              0.007619
Australia                    6215        528            54              0.010748
Israel                      10408       1978            95              0.013998
United Arab Emirates         3360       1561            16              0.016585
South Africa                 2003        348            24              0.017088
Thailand                     2473        304            33              0.018124
Chile                        6501       2030            65              0.019383
South Korea                 10450        213           208              0.020948
Croatia                      1495        313            21              0.021739
Norway                       6314        627           113              0.022145
Malaysia                     4346        684            70              0.022778
Saudi Arabia                 3651       1249            47              0.025451
Russia                      11917       6528            94              0.026768
Pakistan                     4695       1538            66              0.027663
Austria                     13555       1504           319              0.028751
Estonia                      1258        161            24              0.029104
Czechia                      5732       1145           119              0.031096
Finland                      2769        842            48              0.032258
Germany                    122171      22048          2767              0.032643
Japan                        5530       2391            99              0.040064
China                       82941        339          3340              0.040530
Portugal                    15472       4194           435              0.048257
Canada                      22059       6303           557              0.049446
Panama                       2752        951            66              0.050835
Slovenia                     1160        163            45              0.051225
Switzerland                 24551       3451          1002              0.053272
Turkey                      47029      19960          1006              0.055525
Moldova                      1438        574            29              0.059289
```
I think there is good reason to adjust the CFR by age, but it seems from some counter examples, that adjusting it to 4% for the "old" countries might be a big overestimation still. I just wouldn't expect to see developed countries with 1% lagged CFR if that would be the case. 

In other news, I've came across some real data about current ICU occupancy for Germany and for Israel and they both seem pretty close to what the calculations estimate (within 10%-20% percent), which might be a lucky coincidence with all the crude assumptions, but still might be a sign that for other countries it might be of the right order magnitude. (The ICU calculations don't involve that "true CFR" btw, so unrelated to the main point we discussed, but are related to what started the dicussion)

**Comment by HenkPoley** [comment]
Japan's Resolved CFR is more like 12-13%, btw

**Comment by HenkPoley** [comment]
Recent serological tests of blood donors in The Netherlands showed 3% were infected. So maybe.. you are actually lowballing the figures 😉 🤷‍♂️

You can read it in English in [this excellent scientific journal](https://www.dailymail.co.uk/health/article-8224915/Dutch-study-suggests-3-population-coronavirus-antibodies.html) ;)

In dutch, from the horses mouth (Sanquin): https://www.sanquin.nl/artikelen/nieuwsberichten/2020/04/sanquin-ongeveer-3-van-donors-heeft-corona-antistoffen

**Comment by artdgn** [comment]
@HenkPoley Thanks for that. 
1. On one hand, It's a bit hard to know what the biases in that study are (are these random blood donors, or self-selected) without the paper. Also publication bias may have an effect here (especially if it made to popular-press). 
2. On the other hand, I have seen that mentioned in other places that in a lot of Europe's heavily hit countries there is under-attributing of deaths to COVID as patients who do not make it to the hospital (e.g. die in the nursing home), aren't eventually tested and so are not counted. So it's possible that it's 3%, but that may mean that there could be 2-3 times more deaths than actually reported. It is possible, but I'm not sure how to try and account for that.

**Comment by HenkPoley** [comment]
There is a graph of "over-mortality" in The Netherlands: https://www.rivm.nl/monitoring-sterftecijfers-nederland

Yes, I'm also concerned that the test would pick up 10% (of the ones who tested positive now) in older blood samples.

Viewed from the outside it might as well be that nobody had anti-bodies, and they essentially went on a "p-hacking fishing expedition" that showed people sometimes have anti-bodies for the common cold coronavirus.

**Comment by HenkPoley** [comment]
Yesterday around 16:00 UTC+2 Ernst Kuipers (an ICU official, of 'Landelijk Netwerk Acute Zorg') told that current models (I guess by RIVM.nl) tell that The Netherlands is at 4% total affected population. Which is roughly in line with your model.

**Comment by artdgn** [comment]
I've also recently updated the lower bound CFR to be [0.72%](https://www.cebm.net/covid-19/global-covid-19-case-fatality-rates/), which bumped up a lot of European countries to quite high percentages. 
I wonder how much of a big factor [excess deaths](https://www.euromomo.eu/graphs-and-maps)  could be hiding, might even be 2x or so..

**Comment by HenkPoley** [comment]
In The Netherlands the excess mortality was (is?) 2x of the known COVID-19 mortality, yes.

**Comment by HenkPoley** [comment]
FYI, The Netherlands just published their own dashboard: https://coronadashboard.rijksoverheid.nl/
